components/kyma-environment-broker/internal/storage/dbsession/read.go

```diff
@@ -73,6 +73,24 @@ func (r readSession) FindAllInstancesForRuntimes(runtimeIdList []string) ([]inte
 	return instances, nil
 }
 
+func (r readSession) FindAllInstancesForSubAccounts(subAccountslist []string) ([]internal.Instance, dberr.Error) {
+	var instances []internal.Instance
+
+	err := r.session.
+		Select("*").
+		From(postsql.InstancesTableName).
+		Where("sub_account_id IN ?", subAccountslist).
+		LoadOne(&instances)
+
+	if err != nil {
+		if err == dbr.ErrNotFound {
+			return []internal.Instance{}, nil
```

Comment:
please return error here if there is no special reason not to do so

```suggestion
			return []internal.Instance{}, dberr.NotFound("Cannot find Instances for subaccount ID list: '%v'", runtimeIdList)
```

Reply:
In this situation, "not found" case in not an error, there could be empty list of instances for deleted subaccounts from events. If it occurs then no one instance will be remove.